plugins/plugin-bitbucket/codenvy-plugin-bitbucket-ext-bitbucket/src/main/java/org/eclipse/che/ide/ext/bitbucket/server/Bitbucket.java

```diff
@@ -14,65 +14,68 @@
  */
 package org.eclipse.che.ide.ext.bitbucket.server;
 
-import static java.net.URLDecoder.decode;
-import static java.net.URLEncoder.encode;
-import static org.eclipse.che.commons.json.JsonHelper.toJson;
-import static org.eclipse.che.commons.json.JsonNameConventions.CAMEL_UNDERSCORE;
-import static org.eclipse.che.ide.MimeType.APPLICATION_FORM_URLENCODED;
-import static org.eclipse.che.ide.MimeType.APPLICATION_JSON;
 import static org.eclipse.che.ide.ext.bitbucket.shared.Preconditions.checkArgument;
 import static org.eclipse.che.ide.ext.bitbucket.shared.StringHelper.isNullOrEmpty;
-import static org.eclipse.che.ide.rest.HTTPHeader.ACCEPT;
-import static org.eclipse.che.ide.rest.HTTPHeader.AUTHORIZATION;
-import static org.eclipse.che.ide.rest.HTTPHeader.CONTENT_TYPE;
-import static org.eclipse.che.ide.rest.HTTPMethod.GET;
-import static org.eclipse.che.ide.rest.HTTPMethod.POST;
-import static org.eclipse.che.ide.rest.HTTPStatus.CREATED;
-import static org.eclipse.che.ide.rest.HTTPStatus.OK;
 
+import org.eclipse.che.api.auth.oauth.OAuthAuthorizationHeaderProvider;
 import org.eclipse.che.api.auth.oauth.OAuthTokenProvider;
-import org.eclipse.che.api.auth.shared.dto.OAuthToken;
 import org.eclipse.che.api.core.ServerException;
-import org.eclipse.che.commons.env.EnvironmentContext;
-import org.eclipse.che.commons.json.JsonHelper;
-import org.eclipse.che.commons.json.JsonParseException;
-import org.eclipse.che.dto.server.DtoFactory;
+import org.eclipse.che.api.core.rest.HttpJsonRequestFactory;
 import org.eclipse.che.ide.ext.bitbucket.shared.BitbucketPullRequest;
-import org.eclipse.che.ide.ext.bitbucket.shared.BitbucketPullRequestsPage;
-import org.eclipse.che.ide.ext.bitbucket.shared.BitbucketRepositoriesPage;
 import org.eclipse.che.ide.ext.bitbucket.shared.BitbucketRepository;
 import org.eclipse.che.ide.ext.bitbucket.shared.BitbucketRepositoryFork;
 import org.eclipse.che.ide.ext.bitbucket.shared.BitbucketUser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import javax.inject.Named;
+import javax.inject.Singleton;
 import javax.validation.constraints.NotNull;
 import javax.inject.Inject;
-import java.io.BufferedWriter;
-import java.io.ByteArrayOutputStream;
 import java.io.IOException;
-import java.io.InputStream;
-import java.io.OutputStreamWriter;
-import java.net.HttpURLConnection;
-import java.net.URL;
-import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
 
 /**
  * Contains methods for retrieving data from BITBUCKET and processing it before sending to client side.
  *
  * @author Kevin Pollet
+ * @author Igor Vinokur
  */
+@Singleton
 public class Bitbucket {
```

Comment:
Mb you can remove this class at all and solve issue by two providers:
provider for bitbucket endpoint
provider for bitbucket connection.
Mb event by only by one provider for `bitbucket connection`. But I don't like sending http request in constructor body.
@skabashnyuk WDYT?